ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/task/AbstractEthTask.java

```diff
@@ -69,44 +71,39 @@ public double stopTimer() {
 
   @Override
   public final CompletableFuture<T> run() {
-    if (result.compareAndSet(null, new CompletableFuture<>())) {
+    if (!result.isCancelled() && started.compareAndSet(false, true)) {
```

Comment:
`isDone` instead of `isCancelled`?  Can we re-execute something that completed normally?  `isDone` catches _any_ termination.  

Also, perhaps move the started before the cancelled/done check?

Reply:
You're right, I think it should be `isDone`.